src/storage/src/source/postgres/replication.rs

```diff
@@ -171,97 +172,98 @@ pub(crate) fn render<G: Scope<Timestamp = MzOffset>>(
                 return Ok(());
             }
 
-            // Determine the slot lsn.
             let connection_config = connection
                 .connection
                 .config(&*context.secrets_reader)
                 .await?
                 .tcp_timeouts(config.params.pg_source_tcp_timeouts.clone());
 
-            let slot = &connection.publication_details.slot;
-            let replication_client = connection_config.connect_replication(
+            let client = connection_config.connect(
+                "replication metadata",
                 &context.ssh_tunnel_manager,
             ).await?;
-            super::ensure_replication_slot(&replication_client, slot).await?;
-            let slot_lsn = super::fetch_slot_resume_lsn(&replication_client, slot).await?;
+            if let Err(err) = ensure_publication_exists(&client, &connection.publication).await? {
+                // If the publication gets deleted there is nothing else to do. These errors are
+                // not retractable.
+                for &oid in table_info.keys() {
+                    // We must emit this error at a definite LSN which ideally would be the LSN at
+                    // which the `DROP PUBLICATION` action was written to in the upstream database.
+                    // Unfortunately we don't have a way to learn that LSN and so we must choose an
+                    // LSN out of thin air that is guaranteed to not invalidate the definite
+                    // decisions previously made by this operator. We therefore always pick
+                    // `u64::MAX`, which will (in practice) never conflict any previously revealed
+                    // portions of the TVC.
+                    let update = ((oid, Err(err.clone())), MzOffset::from(u64::MAX), 1);
+                    data_output.give(&data_cap_set[0], update).await;
+                }
+                return Ok(());
+            }
 
-            let resume_upper = Antichain::from_iter(
+            // This is an ugly hack that we need to do until because the `build_fallible`
+            // constructor (which we use here) is unsafe in the presence of retained input
+            // capabiltities (which we also use for the rewind requests). The mutable antichain is
+            // initialized with one copy of the minimum capability which represents the initial
+            // capability of this operator.
+            let mut fake_capability = MzOffset::minimum();
+            let mut fake_progress = MutableAntichain::new_bottom(MzOffset::minimum());
```

Comment:
I don't really like the name `fake_` here, what about `synthesized` or something?

Reply:
sure, I'll change the name